client/src/unifycr-sysio.c

```diff
@@ -325,28 +325,81 @@ int UNIFYCR_WRAP(remove)(const char *path)
 
 int UNIFYCR_WRAP(stat)(const char *path, struct stat *buf)
 {
+    int ret = 0;
+
     DEBUG("stat was called for %s....\n", path);
     if (unifycr_intercept_path(path)) {
+        if (!buf) {
```

Comment:
If `buf` is not valid, we should set `EFAULT`. I made some changes for this in existing `stat(2)` implementations.